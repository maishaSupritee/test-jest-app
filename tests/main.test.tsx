import Main from "@/app/components/main";
import { render, screen } from "@testing-library/react";

test("main component matches snapshot", () => {
  const { asFragment } = render(<Main />);
  expect(asFragment()).toMatchSnapshot();
});

test("all static elements are rendered correctly", () => {
  render(<Main />);
  const header = screen.getByRole("heading", { name: /jest test/i });
  const description = screen.getByText(/Let's try writing some unit tests!/i);
  const image = screen.getByAltText("Test Image");

  expect(header).toBeInTheDocument();
  expect(description).toBeInTheDocument();
  expect(image).toBeInTheDocument();
});

test("test button and link elements work properly", () => {
  render(<Main />);

  const link = screen.getByTestId("pexels-link");
  const button = screen.getByRole("button", {
    name: /go to secondary section/i,
  });
  const buttonLink = button.querySelector("a");

  expect(link).toBeInTheDocument();
  window.open = jest.fn(); // Mock window.open to prevent actual navigation
  link.click();
  expect(window.open).toHaveBeenCalledWith(
    "https://www.pexels.com/photo/time-displayed-on-top-of-a-building-1824273/",
    "_blank"
  );

  expect(button).toBeInTheDocument();
  expect(buttonLink).toHaveAttribute("href", "#secondary");
});
