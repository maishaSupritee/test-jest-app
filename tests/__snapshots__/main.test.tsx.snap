// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`main component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center gap-4"
  >
    <h1
      class="pt-10 text-4xl font-bold"
    >
      Jest Test
    </h1>
    <p
      class=" text-lg"
    >
      Let's try writing some unit tests!
    </p>
    <img
      alt="Test Image"
      data-nimg="1"
      decoding="async"
      height="100"
      loading="lazy"
      src="/_next/image?url=%2Ftest-image.jpg&w=828&q=75"
      srcset="/_next/image?url=%2Ftest-image.jpg&w=640&q=75 1x, /_next/image?url=%2Ftest-image.jpg&w=828&q=75 2x"
      style="color: transparent;"
      width="400"
    />
    <p
      class="text-sm text-gray-500"
    >
      Photo taken from:
      <a
        class="text-blue-500 hover:underline ml-2"
        data-testid="pexels-link"
        href="https://www.pexels.com/photo/time-displayed-on-top-of-a-building-1824273/"
      >
        pexels
      </a>
    </p>
    <button
      class="px-6 py-3 bg-blue-500 text-white rounded-lg hover:bg-blue-600 transition-colors duration-200 mt-4"
    >
      <a
        href="#secondary"
      >
        Go to Secondary Section
      </a>
    </button>
  </div>
</DocumentFragment>
`;
